libfs/archive_util.go

```diff
@@ -27,3 +29,37 @@ func BranchNameFromArchiveRefDir(dir string) (libkbfs.BranchName, bool) {
 
 	return libkbfs.MakeRevBranchName(kbfsmd.Revision(rev)), true
 }
+
+// RevFromTimeString converts a time string (in any supported golang
+// date format) to the earliest revision number with a server
+// timestamp greater or equal to that time.  Ambiguous dates are
+// parsed in MM/DD format.
+func RevFromTimeString(
+	ctx context.Context, config libkbfs.Config, h *libkbfs.TlfHandle,
+	timeString string) (kbfsmd.Revision, error) {
+	t, err := dateparse.ParseAny(timeString)
```

Comment:
Why? I don't want it to error on ambiguous dates like "03/04/2018", I'd rather have it do something deterministic, since there's no way to communicate back to the user what the error is.

Reply:
It's deterministic from a developer perspective, but confusing when you consider that date format is different across the world. If we use `Parse` and prefer the US format, a non-US user may automatically think "03/04/2018" means April 3rd, 2018. It's hard to communicate to user that we are preferring mm/dd/yyyy for the same reason it's hard to communicate what the error is.

Reply:
FWIW I'm always hesitant to enter a `/` separated date unless there's a clear instruction about whether it's mm/dd or dd/mm. In CLI I do "yyyy-mm-dd" whenever possible.

Reply:
I understand that, but I think in this case ambiguity is better than a flat out error.  In the GUI version, this won't be an issue as people will be using a date picker or something like that, and it's not an issue if you're copying in the output of `date` for example, so I'm going to leave it as is for now.  We'll document it in the understanding_kbfs page, which is probably the only way anyone will even know about this feature.

Reply:
Cool; thanks! I think yyyy-m-dd also works, and it's the same number of characters:

```
%  go run main.go '2018-7-27 22:05'
2018-07-27 22:05:00 +0000 UTC <nil>
%  cat main.go
package main

import (
	"fmt"
	"os"

	"github.com/araddon/dateparse"
)

func main() {
	fmt.Println(dateparse.ParseStrict(os.Args[1]))
}
```

Reply:
`2018-7-27` is still two more characters than `7/27/18`.  I'll figure something out.

Reply:
Oh it's yy! OK ... thanks!